gms/inet_address.cc

```diff
@@ -23,7 +23,7 @@ static_assert(std::is_nothrow_default_constructible_v<gms::inet_address>);
 static_assert(std::is_nothrow_copy_constructible_v<gms::inet_address>);
 static_assert(std::is_nothrow_move_constructible_v<gms::inet_address>);
 
-future<gms::inet_address> gms::inet_address::lookup(sstring name, opt_family family, opt_family preferred) {
+future<gms::inet_address> gms::inet_address::lookup(const sstring& name, opt_family family, opt_family preferred) {
```

Comment:
std::string_view maybe?

Reply:
but it'd defeat the purpose of this change, as `gms::inet_address::lookup()` ends up calling into `seastar::net::inet_address::parse_numerical()` which expects an `sstring`, and `seastar::net::inet_address::parse_numerical()` uses [inet_pton(3) ](https://man7.org/linux/man-pages/man3/inet_pton.3.html), which in turn expects an nul-terminated string, but `string_view` does not guarantee this. this this is a dead end.

Reply:
after a second thought, i think the better alternative is to pass `std::move(name)` , a little bit overkill. but that's what `seastar::net::dns::get_host_by_name()` expects, i guess.